File: 1. 단계별로_풀어보기/20~29/22. 누적 합/3. 16139.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const S = inputArr[0].trim()
const queryArr = inputArr.slice(2).map(row => {
  const [c, i, j] = row.trim().split(' ')
  return [c, +i, +j]
})

// 결과 배열
const sumArr = Array.from({length: 26}, () => Array.from({length: S.length + 1}).fill(0))

// 'a' 의 아스키코드 값
const a = 'a'.charCodeAt(0)

// Bottom-Up 방식으로 sumArr 채우기
for (let i = 1; i <= S.length; i++) {
  for (let c = 0; c < 26; c++) {
    sumArr[c][i] = sumArr[c][i - 1]
  }
  const nowC = S.charCodeAt(i - 1) - a
  sumArr[nowC][i] += 1
}

// 출력 변수
let resultStr = ''

queryArr.forEach(query => {
  const [c, i, j] = query
  const cASCII = c.charCodeAt(0) - a

  resultStr += sumArr[cASCII][j + 1] - sumArr[cASCII][i] + '\n'
})

// 출력
process.stdout.write(`${resultStr}`)

File: 1. 단계별로_풀어보기/20~29/22. 누적 합/3. 16139.py
# PyPy3 으로 돌려야 100점이 나온다.

import sys

input = sys.stdin.readline
output = sys.stdout.write

S = list(input().strip())
Q = int(input())
queryArr = list(input().strip().split() for i in range(Q))

lenS = len(S) # 중복연산 방지
a = ord('a')

sumArr = list(list(0 for i in range(lenS + 1)) for j in range(26))

firstC = queryArr[0][0]
sumArr[ord(firstC) - a][0] = 1

for s in range(1, lenS + 1):
  for cIdx in range(26):
    sumArr[cIdx][s] = sumArr[cIdx][s - 1]
  
  c = S[s - 1]
  sumArr[ord(c) - a][s] += 1

for query in queryArr:
  c, i, j, = query
  i = int(i)
  j = int(j)

  queryResult = sumArr[ord(c) - a][j + 1] - sumArr[ord(c) - a][i]
  output(f"{queryResult}\n")